k8s-operator/sessionrecording/hijacker.go

```diff
@@ -45,7 +49,7 @@ var (
 	counterSessionRecordingsUploaded = clientmetric.NewCounter("k8s_auth_proxy_session_recordings_uploaded")
 )
 
-func New(ts *tsnet.Server, req *http.Request, who *apitype.WhoIsResponse, w http.ResponseWriter, pod, ns string, proto protocol, addrs []netip.AddrPort, failOpen bool, connFunc RecorderDialFn, log *zap.SugaredLogger) *Hijacker {
+func New(ts *tsnet.Server, req *http.Request, who *apitype.WhoIsResponse, w http.ResponseWriter, pod, ns string, proto Protocol, addrs []netip.AddrPort, failOpen bool, connFunc RecorderDialFn, log *zap.SugaredLogger) *Hijacker {
```

Comment:
lol this function signature is amazing. maybe have the caller create the struct directly? you might have to export the fields but then you get nicer docs

Reply:
heh, I am refactoring it in a follow-up https://github.com/tailscale/tailscale/pull/12965/commits/d8dc42b8560dc693a3140f331579801734ab306e#diff-ce264fec78adcf2abf1810777756d8abcc179ec8a6df839b00687209d383e863R257-R269

Reply:
I can pull that in here